### PR TITLE
Update Go version, remove vendor, and misc. upgrades

### PR DESCRIPTION
**Pull request by aiyengar2**
As part of the changes necessary for adding the `wins cli proxy` command, a ton of small changes to wins w.r.t. the Go version and dependencies and build workflow needed to be changed first.

I've split all of these non-proxy related changes from https://github.com/rancher/wins/pull/14 to this PR in order to make it easier to review.

Also takes care of escaped quote support for https://github.com/rancher/wins/issues/9.

Related Issue: https://github.com/rancher/wins/issues/13, https://github.com/rancher/wins/issues/9

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # IDE
 .idea/
+.vscode/
```
Comment (aiyengar2) [inline .gitignore:3]: #### Note

When using VSCode / VSCodium to develop on a Mac, you need to set the following on `vscode/settings.json` to get the Go plugin working correctly:

```yaml
{
    "go.toolsEnvVars": {
        "GOOS" : "windows"
    },
}
```

File: Dockerfile.dapper [hunk 2]
```diff
@@ -73,7 +74,7 @@ RUN pushd c:\; \
     \
     Write-Host 'Installing gcc ...'; \
     $PACKAGES = 'mingw64-x86_64-gcc-core,mingw64-x86_64-gcc-g++'; \
-    Start-Process -NoNewWindow -Wait -FilePath 'c:\cygwin64\setup.exe' -ArgumentList ('-q -d -X -s {0} -D -L -R {1} -l {2} -P {3}' -f 'http://cygwin.mirrors.pair.com/', 'C:/cygwin64', $env:TEMP, $PACKAGES); \
+    Start-Process -NoNewWindow -Wait -FilePath 'c:\cygwin64\setup.exe' -ArgumentList ('-q -d -X -s {0} -D -L -R {1} -l {2} -P {3}' -f 'https://mirror.clarkson.edu/cygwin/', 'C:/cygwin64', $env:TEMP, $PACKAGES); \
```
Comment (aiyengar2) [inline Dockerfile.dapper:77]: We can use any mirror here that is listed in https://cygwin.com/mirrors.html.

File: Dockerfile.dapper [hunk 3]
```diff
@@ -82,39 +83,9 @@ RUN pushd c:\; \
...
-    \
-    Write-Host 'Expanding ...'; \
-    Expand-Archive -Path c:\patch.zip -DestinationPath c:\patch; \
-    \
-    Write-Host 'Cleaning ...'; \
-    Remove-Item -Force -Recurse -Path c:\patch.zip; \
     \
-    Write-Host 'Updating PATH ...'; \
-    [Environment]::SetEnvironmentVariable('PATH', ('c:\patch\bin\;{0}' -f $env:PATH), [EnvironmentVariableTarget]::Machine); \
+    Get-ChildItem -Path C:\cygwin64 -Recurse -Attributes ReparsePoint | Remove-Item; \
```
Comment (aiyengar2) [inline Dockerfile.dapper:87]: https://github.com/moby/moby/issues/41058#issuecomment-721156197

File: cmd/cmds/flags/list_value_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package flags
```
Comment (aiyengar2) [inline cmd/cmds/flags/list_value_test.go:1]: @luthermonson 

```bash
arvindiyengar: ~/Rancher/wins/src/github.com/rancher/wins/cmd/cmds/flags 
$ go test
PASS
ok  	github.com/rancher/wins/cmd/cmds/flags	0.134s
```

File: Dockerfile.dapper (hunk 2 above)
Comment (aiyengar2) [reply Dockerfile.dapper:78]: add this ^ as a comment on the Dockerfile itself

File: cmd/cmds/flags/list_value_test.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	if err != nil {
+		t.Error(err)
+	}
+	// Check outputted values
+	expectedList := []string{"RANCHER=hello", "WINS=world"}
+	err = fmt.Errorf("Failed to parse list value: expected %v, got %v", expectedList, valList)
+	for i, expected := range expectedList {
+		if valList[i] != expected {
+			t.Error(err)
+			return
```
Comment (nickgerace) [inline cmd/cmds/flags/list_value_test.go:20]: Is this return intentional?

File: cmd/cmds/flags/list_value_test.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	if err != nil {
+		t.Error(err)
+	}
+	// Check outputted values
+	expectedList := []string{"NICK=bye", "LUTHER=\"hello\"", "RANCHER=\"hello world\"", "WINS=world"}
+	err = fmt.Errorf("Failed to parse list value: expected %v, got %v", expectedList, valList)
+	for i, expected := range expectedList {
+		if valList[i] != expected {
+			t.Error(err)
+			return
```
Comment (nickgerace) [inline cmd/cmds/flags/list_value_test.go:37]: t.Fatal() might be better here

**Review by nickgerace — APPROVED** [review]
LGTM but @luthermonson pressured me to approve.

**Review by luthermonson — APPROVED** [review]

**Pull request merged**